### PR TITLE
KOGITO-2099: Generate a SVG diagram automatically on each BPMN/DMN diagrams save

### PR DESCRIPTION
**Pull request by ederign**
To make the use of VSCode more friendly for RHPAM users (especially on the integration with Kie Server), the idea is to provide a configurable way to generate the SVG on each save of the BPMN/DMN diagram and provide a similar capability of business central.

The following commands were added: 

<code>
"kogito.dmn.runOnSave": "extension.kogito.silentlyGenerateSvgDmn",

"kogito.bpmn.runOnSave": "extension.kogito.silentlyGenerateSvgBpmn"
</code>

Whenever the user config those commands, we will auto-generate (silently) the SVG for BPMN and DMN diagrams. (PMML extension was not affected).

For vscode-extension-pack-kogito-kie-editors, I've configured the respective parameters as default.

[Feature Video](https://p192.p3.n0.cdn.getcloudapp.com/items/GGupbBwl/f9ed294c-2eeb-4918-b3fc-f74f1b50fabf.gif?source=client&v=f1da3017eff42b749d56fdabbd208568)

### CODE REVIEW

File: packages/vscode-extension-pack-kogito-kie-editors/package.json [hunk 1]
```diff
@@ -28,6 +28,27 @@
...
+          "default": "extension.kogito.silentlyGenerateSvg",
+          "description": "Execute a command on each save operation of the BPMN file."
+        },
+        "kogito.dmn.runOnSave": {
+          "type": [
+            "string",
+            "null"
+          ],
+          "default": "extension.kogito.silentlyGenerateSvg",
+          "description": "Execute a command on each save operation of the BPMN file"
```
Comment (tomasdavidorg) [inline packages/vscode-extension-pack-kogito-kie-editors/package.json]:
```suggestion
          "description": "Execute a command on each save operation of the DMN file"
```

**Comment by tomasdavidorg** [comment]
@ederign I bit confused what is the default value for `kogito.bpmn.runOnSave` and `kogito.bpmn.runOnSave`. In the VS code settings, there is `extension.kogito.silentlyGenerateSvg` but in the video there is `extension.kogito.generateSVGOnSaveBpmn` and `extension.kogito.generateSVGOnSaveDmn`.

Do I understand correctly that for both `kogito.bpmn.runOnSave` and `kogito.bpmn.runOnSave`:
-  if I have empty string `""` then nothing is executed after save
-  if I have `extension.kogito.silentlyGenerateSvg` then the SVG is generated silently after each save?
-  if I have `extension.kogito.getPreviewSvg` then the SVG is generated after each save and notification is displayed?

**Review by tiagobento — APPROVED** [review]

**Comment by ederign** [comment]
Hi @tomasdavidorg, thanks for the review, replies:

> if I have empty string "" then nothing is executed after save

On the BPMN and DMN extension yes. Nothing should also happen also in the absence of the property.
On PMML extension, nothing should be changed.

> if I have extension.kogito.silentlyGenerateSvg then the SVG is generated silently after each save?
For DMN, you need to define the key: kogito.dmn.runOnSave=extension.kogito.silentlyGenerateSvg
For BPMN, kogito.bpmn.runOnSave=extension.kogito.silentlyGenerateSvg


> if I have extension.kogito.getPreviewSvg then the SVG is generated after each save and notification is displayed?

That is correct.


Also, on our 'test' extension (packages/vscode-extension-pack-kogito-kie-editors/), I've configured the parameters to be default. So on each save, we should have the svg generated for both bpmn and dmn editor.s

**Review by tomasdavidorg — APPROVED** [review]
Thanks @ederign.

**Comment by manstis** [comment]
IIRC DMN did not have the SVG generated on save; it was only BPMN. 

We had to make a change on DMN to prevent it from happening as it was not desired.

@karreiro can you please confirm/deny/advise?

**Comment by karreiro** [comment]
Hi @manstis, IIUC the SVG mechanism exists in the DMN side as well to support this kind of use case:

Screenshot 2021-09-24 at 09 50 09

Am I missing something?

**Comment by manstis** [comment]
@karreiro no, no.. I am just out of date.

I remembered that in Business Central we had to disable the auto-SVG generation.

Times have obviously moved on.

**Pull request merged**